crates/libcontainer/src/container/tenant_builder.rs

```diff
@@ -148,13 +149,13 @@ impl TenantContainerBuilder {
         let mut notify_socket = NotifySocket::new(notify_path);
         notify_socket.notify_container_start()?;
 
-        close(write_end).map_err(LibcontainerError::OtherSyscall)?;
+        close(write_end.as_raw_fd()).map_err(LibcontainerError::OtherSyscall)?;
```

Comment:
Since write_end is `OwnedFd`, why you do not close it by `drop(write_end)`? 

```rust
#[stable(feature = "io_safety", since = "1.63.0")]
impl Drop for OwnedFd {
    #[inline]
    fn drop(&mut self) {
        unsafe {
            ...
            let _ = libc::close(self.fd);
            ...
```
https://doc.rust-lang.org/nightly/src/std/os/fd/owned.rs.html#174-186

Reply:
You beat me to it :joy: https://github.com/containers/youki/pull/2728#discussion_r1547168975